New design (UI) - appbar color

JorisBodin opened the pull request:
Hello,

I finished changing the color of the appbar.
I also changed the progressbar to match the new design.

![newAppBar](https://user-images.githubusercontent.com/2796439/80232503-4d925800-8655-11ea-8673-4eaa673f7ac4.gif)

Next week I'll continue on the appbar. I will change in the pictures view and search. When I'm done I'll change its style.

See all futur design: https://github.com/nextcloud/android/issues/5207#issuecomment-617270183

JorisBodin commented:
@jancborchardt The "Search" placeholder has not changed. It's the native android. It should change :)

ezaquarii commented on src/main/java/com/owncloud/android/ui/activity/ToolbarActivity.java:
```diff
@@ -166,25 +160,14 @@ protected final void hideInfoBox() {
      * Change the visibility for the toolbar's progress bar.
      *
-     * @param visibility visibility of the progress bar
+     * @param isVisibility visibility of the progress bar
```
While here, we could actually correct the splling so it's not wonky: `isVisible`.

JorisBodin replied:
For the next PR 😁

AndyScherzinger commented:
Looks good to me 👍

Great work @JorisBrodin 🏆

AndyScherzinger commented:
@tobiasKaminsky what do you think how to much forward branching wise?
I'd be happy to see the UI changes going to master but the whole redesign is unlikely completed for 3.12. So we would have to create 3.12 rather soon and live with a small feature release 3.12 and see that the UI is done for 3.13.

What do you all think? @ezaquarii @JorisBrodin @Shagequi, I think we should make moving this forward as easy as possible where a separate branch would create a update hell regarding stating up to date with Master :/

JorisBodin commented:
@AndyScherzinger 3.12 When is it scheduled?
Cause I think we're done for May 7

ostasevych commented:
What about dark theme? Now it is not shown well, when some texts are also dark on the dark background?

JorisBodin commented:
Hello @ostasevych
Do you have a exemple?

AndyScherzinger commented:
>  3.12 When is it scheduled?

@JorisBodin see #5644

@tobiasKaminsky according to schedule today would be 3.12. RC1, so we would be on time for 3.13 with this (I guess?)

JorisBodin commented:
> > 3.12 When is it scheduled?
> 
> @JorisBodin see #5644

I think the current modification of the appbar can be proposed in 3.12 :)

I modified the variable as suggested by @ezaquarii
And I also changed the ActionMode colors.

JorisBodin commented:
Improve item selected background 
Before
![Screenshot_1587990422](https://user-images.githubusercontent.com/2796439/80372207-72740e80-8893-11ea-80f1-9f8928200210.png)
After
![Screenshot_1587990333](https://user-images.githubusercontent.com/2796439/80372219-76a02c00-8893-11ea-880a-16421a074e5a.png)

tobiasKaminsky commented:
@JorisBodin there are conflicts?
Hopefully I did not created them…

Regarding time frame:
Today I branched off 3.12, which will be released in ~2 weeks.
So we have enough time to get all our design changes in 3.13 :+1:

JorisBodin commented:
@tobiasKaminsky The merge is now ok

I found that it was the wrong color of the back button in preview mode.
![Screenshot_1588081058](https://user-images.githubusercontent.com/2796439/80493867-3e1d5280-8966-11ea-9200-2a68ad24e112.png)

I'll fix it later.

And https://github.com/nextcloud/android/pull/5951 doesn't work, so I took the liberty of removing it
![Screenshot_1588081352](https://user-images.githubusercontent.com/2796439/80494412-ed5a2980-8966-11ea-8b3c-ff7606cc0641.png)

tobiasKaminsky commented on src/main/java/com/owncloud/android/ui/activity/FileDisplayActivity.java:
```diff
@@ -2227,7 +2227,7 @@ public void run() {
                                 );
 
-                                setIndeterminate(true);
+                                showProgressBar(true);//okkk
```
Please remove this comment

JorisBodin replied:
Oupsi 🙈

tobiasKaminsky commented on src/main/java/com/owncloud/android/utils/ThemeUtils.java:
```diff
@@ -224,17 +218,18 @@ public static int fontColor(Context context) {
-    public static boolean lightTheme(Context context) {
-        int primaryColor = primaryColor(context);
-        float[] hsl = colorToHSL(primaryColor);
+    public static boolean lightTheme(Context context, int color) {
```
This one now does not use context any longer.
Please remove it and change doc.

tobiasKaminsky commented on src/main/java/com/owncloud/android/utils/ThemeUtils.java:
```diff
@@ -244,6 +239,22 @@ public static boolean darkTheme(Context context) {
         return hsl[INDEX_LUMINATION] <= 0.55;
     }
 
+    public static int primaryAppbarColor(Context context) {
```
Is this background?
Then maybe appBarPrimaryBackgroundColor

JorisBodin replied:
It's a point of view. But it's ok :)

tobiasKaminsky commented on src/main/java/com/owncloud/android/utils/ThemeUtils.java:
```diff
@@ -244,6 +239,22 @@ public static boolean darkTheme(Context context) {
+        return ContextCompat.getColor(context, R.color.appbar);
+    }
+
+    public static int fontAppbarColor(Context context) {
```
appBarPrimaryFontColor

tobiasKaminsky commented on src/main/java/com/owncloud/android/utils/ThemeUtils.java:
```diff
@@ -244,6 +239,22 @@ public static boolean darkTheme(Context context) {
+        return ContextCompat.getColor(context, R.color.fontAppbar);
+    }
+
+    public static int fontSecondaryAppbar(Context context) {
```
appBarSecondaryFontColor

tobiasKaminsky replied:
(my idea is to first have the "where" and then "what".

tobiasKaminsky commented:
Some minor code formatting changes.
@jancborchardt have you tested the apk and are you fine with the design so far?

tobiasKaminsky commented:
Merged master again into it.
This should now work, and then can be merged.

AndyScherzinger commented:
One minor issue that should be fixed (maybe in another PR?) is the intermediate bar which when shown/hidden makes the content jump, this didn't happen before.

Is this appbar shown anywhere where the loading needs to be displayed? Asking sind the search bar would rather use the circular spinner and maybe we should just switch to that one only, since we use it on other screens like activities or notifications.
